firefox_puppeteer/ui/toolbars.py

```diff
@@ -0,0 +1,337 @@
+
+        See the :class:`~ui.toolbars.AutocompleteResults` reference.
+        """
+
+    @use_class_as_property('ui.toolbars.IdentityPopup')
+    def identity_popup(self):
+        """Provides utility members for accessing and manipulationg the
+        locationbar.
+
+        See the :class:`~ui.toolbars.IdentityPopup` reference.
+        """
+
+    @property
+    def urlbar(self):
+        """:returns: The urlbar element.
+        """
+        return self.marionette.find_element(By.ID, 'urlbar')
+
+    @property
+    def reload_button(self):
+        return self.marionette.find_element(By.ID, 'urlbar-reload-button')
+
+    @property
+    def favicon(self):
+        return self.marionette.find_element(By.ID, 'page-proxy-favicon')
+
+    @property
+    def stop_button(self):
+        return self.marionette.find_element(By.ID, 'urlbar-stop-button')
+
+    @property
+    def urlbar_input(self):
+        """:returns: The urlbar_input element
+        """
+        return self.urlbar.find_element('anon attribute', {'anonid': 'input'})
+
+    @property
+    def history_drop_marker(self):
+        return self.urlbar.find_element('anon attribute', {'anonid': 'historydropmarker'})
+
+    @property
+    def contextmenu(self):
+        """:returns: The urlbar contextmenu element.
+        """
+        parent = self.urlbar.find_element('anon attribute', {'anonid': 'textbox-input-box'})
+        return parent.find_element('anon attribute', {'anonid': 'input-box-contextmenu'})
+
+    @property
+    def notification_popup(self):
+        return self.marionette.find_element(By.ID, "notification-popup")
+
+    @property
+    def value(self):
+        """:returns: The currently displayed value of the urlbar.
+        """
+        return self.urlbar.get_attribute('value')
+
+    def get_contextmenu_entry(self, prop, subtype):
+        """
+        :param subtype: The type of entry to retrieve.
+        :returns: The urlbar contextmenu entry.
+        """
+        entries = self.contextmenu.find_elements('css selector', 'menuitem')
+        filter_on = 'cmd_%s' % subtype
+        found = [e for e in entries if e.get_attribute(prop) == filter_on]
+        return found[0] if len(found) else None
+
+    def close_context_menu(self):
+        self.contextmenu.send_keys(Keys.ESCAPE)
+
+    def clear(self):
+        """Clears the contents of the url bar (via the DELETE shortcut).
+
+        Note that the operation is synchronous: it does not return until
+        the value of the url bar is cleared.
+        """
+        self.focus('shortcut')
+        self.urlbar.send_keys(Keys.DELETE)
+        Wait(self.marionette).until(
+            lambda _: self.urlbar.get_attribute('value') == '')
+
+    def load_url(self, url):
+        """Load the specified url in the location bar by synthesized
+        keystrokes.
+        """
+        self.focus('shortcut')
+        self.urlbar.send_keys(url + Keys.ENTER)
+
+    def reload_url(self, evt='shortcut', force=False):
```

Comment:
Instead of `evt` please use `trigger` so we are in sync with the window classes. Also I would keep the name of the method `reload`. I think it is clear enough to understand what it does.